service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java

```diff
@@ -405,4 +399,32 @@ private LocalDateTime calculateExpirationDate(Integer expirationTimeInHours) {
         LocalDateTime now = LocalDateTime.now();
         return now.plusHours(expirationTimeInHours);
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public Boolean hasPassword(String email) {
+        User user = userRepo.findByEmail(email)
+            .orElseThrow(() -> new WrongEmailException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL));
+        return user.getOwnSecurity() != null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setPassword(SetPasswordDto dto, String email) {
+        User user = userRepo.findByEmail(email)
+            .orElseThrow(() -> new WrongEmailException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL));
+        if (Boolean.TRUE.equals(hasPassword(email))) {
```

Comment:
why not just if (hasPassword(email)) ? user.getOwnSecurity() != null; - never returns null